src/fragmentarium/ui/display/Transliteration.tsx

```diff
@@ -201,24 +201,38 @@ const lineComponents: ReadonlyMap<
   ['CompositeAtLine', DisplayDollarAndAtLine]
 ])
 
+const rulingsToNumber: ReadonlyMap<string, number> = new Map([
+  ['SINGLE', 1],
+  ['DOUBLE', 2],
+  ['TRIPLE', 3]
+])
+
 function DisplayRulingDollarLine({
   line,
   container = 'div'
 }: {
   line: Line
   container?: string
 }): JSX.Element {
+  function drawRulings(line: RulingDollarLine): Array<JSX.Element> {
+    const rulingsNumber = rulingsToNumber.get(line.number) as number
+    const items: Array<JSX.Element> = []
+    for (let i = 1; i <= rulingsNumber; i++) {
+      items.push(
+        React.createElement('div', {
+          className: `Transliteration__RulingDollarLine--${line.number.toLowerCase()}`,
+          key: i
+        })
+      )
+    }
+    return items
+  }
```

Comment:
This seems a very convoluted way to create three elements.

- This could be a regular component, instead of returning a list of elements.
- There is no need to use `createElement`, you can just use `<div key=... className=...>`
- There is no need for an intermediate `items` variable. You can use `map` on a range of numbers.
- It does not make a sense to add `Transliteration__RulingDollarLine` element inside a `Transliteration__RulingDollarLine` element. The elements used for each line should have another name, e.g. `Transliteration__ruling`.